### PR TITLE
WW-5184 - Add optional parameter value check to ParametersInterceptor

### PR DESCRIPTION
**Pull request by brianandle**

### CODE REVIEW

**Review by yasserzamani — CHANGES_REQUESTED** [review]
Could you please move these changes to main branch? we won't have Struts 2.5.x releases except for approved security reasons. Anyway following there are some comments that I could suggest in the first place. Thank you for your PR!

File: core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java [hunk 1]
```diff
@@ -177,8 +186,13 @@ protected void setParameters(final Object action, ValueStack stack, HttpParamete
 
         for (Map.Entry<String, Parameter> entry : params.entrySet()) {
             String parameterName = entry.getKey();
-
-            if (isAcceptableParameter(parameterName, action)) {
+            boolean isAcceptableParameter;
+            if(hasParamValuesToExclude() || hasParamValuesToAccept()) {
```
[the remark this answers is not in the document]
Comment (brianandle) [reply core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java]: Done

File: core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java [hunk 2]
```diff
@@ -292,6 +327,28 @@ protected boolean acceptableName(String name) {
...
+     * 
+     * @param name - Name to check
+     * @param value - value to check
+     * @return true if accepted
+     */
+    protected boolean acceptableNameValue(String name, String value) {
+    	boolean accepted = isWithinLengthLimit(name) && !isExcluded(name) && isAccepted(name) 
+    			&& (value == null || value.isEmpty() || (!isParamValueExcluded(value) && isParamValueAccepted(value)));
+        if (devMode && accepted) { // notify only when in devMode
+            LOG.debug("Parameter [{}] was accepted with value [{}] and will be appended to action!", name, value);
```
[the remark this answers is not in the document]
Comment (brianandle) [reply core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java]: Done

File: core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java [hunk 3]
```diff
@@ -335,7 +392,85 @@ protected boolean isExcluded(String paramName) {
...
+	                    LOG.trace("[{}] matches excluded pattern [{}]", value, excludedPattern);
+	                    return true;
+	                }
+	    		}
+	    	}
+    	} else {
+    		// acceptedValuePatterns not defined so anything is allowed
+    		return true;
+    	}
+    	return false;
```
[the remark this answers is not in the document]
Comment (brianandle) [reply core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java:461]: Done

File: core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java [hunk 4]
```diff
@@ -58,6 +64,9 @@ public class ParametersInterceptor extends MethodFilterInterceptor {
     private ValueStackFactory valueStackFactory;
     private ExcludedPatternsChecker excludedPatterns;
     private AcceptedPatternsChecker acceptedPatterns;
+    private Set<Pattern> excludedValuePatterns = null;
+    private Set<Pattern> acceptedValuePatterns = null;
```
[the remark this answers is not in the document]
Comment (yasserzamani) [reply core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java:68]: I see, you're absolutely right, thanks for clarification!

**Review by yasserzamani — APPROVED** [review]
LGTM:+1: thanks again!

**Pull request closed without merging**